src/lib/core/components/visualizations/implementations/HistogramVisualization.tsx

```diff
@@ -451,12 +471,26 @@ function HistogramViz(props: VisualizationProps<Options>) {
       facetVariable,
       facetEntity,
       valueType,
-      // get data when changing independentAxisRange
-      vizConfig.independentAxisRange,
     ])
   );
 
-  const defaultIndependentRange = useDefaultAxisRange(xAxisVariable);
+  const independentAxisMinMax = useMemo(
+    () => histogramDefaultIndependentAxisMinMax(data),
+    [data]
+  );
+
+  const defaultIndependentRange = useDefaultAxisRange(
+    xAxisVariable,
+    vizConfig.independentAxisValueSpec === 'Full'
```

Comment:
@bobular Thank you for your tests and comments 👍 Yes, when implementing the new axis range controls for histrogram x-axis, I realized that actually Full and Auto-zoom options did not mean much. The need to have those two tests for `vizConfig.independentAxisValueSpec === 'Full'` depends on whether we will distinguish Full and Auto-zoom options for histogram x-axis: Case 1) if we don't need to have different values for those two options, then we can simply use old logic `const defaultIndependentRange = useDefaultAxisRange(xAxisVariable);` and remove the new works. Case 2)  If we need to distinguish them, then we need new additions, `independentAxisMinMax` and `defaultIndependentRange` with the two `vizConfig.independentAxisValueSpec === 'Full'` conditions for `useDefaultAxisRange` hook. Note that without the two conditions, both Full and Auto-zoom may lead to data-based min/max, not metadata-based min/max, e.g., GEMS1 Age (for main variable) where metadata-based min/max is [0,62] while data-based min/max is [0,63] with the default bin width of 3.

Which one do you prefer?

Reply:
For me, the following is working fine for Histogram -> GEMS1 -> Age

Full goes 0 to 62, Auto-zoom goes 0 to max-of-my-subset.

```
  const defaultIndependentRange = useDefaultAxisRange(
    xAxisVariable,
    independentAxisMinMax?.min,
    undefined, // never logScale on this axis so no need for minPos
    independentAxisMinMax?.max,
    undefined, // never logScale - I would prefer `false` here
    vizConfig.independentAxisValueSpec
  );
```

Reply:
@bobular Can you please check it out again? If we use it like your listed, then both Full and Auto-zoom goes to [0,63] (with the default bin width of 3) because [independentAxisMinMax.min, independentAxisMinMax.max] = [0,63] and Full option will compute max among metadata-based, data-based, etc. I am getting both [0,63] - just in case, please hard refresh your browser 😃 (CTRL + F5 key in Chrome) or check it at private mode :)